### PR TITLE
Rename yang enum BAREFOOT_TOFINO to GENERIC_BAREFOOT_TOFINO

### LINKED ISSUE #442: Use GENERIC
https://github.com/stratum/stratum/blob/9f5bd2b285badbef11e81eca6c31d4a3c4342843/stratum/public/model/stratum.yang#L66

### PR DESCRIPTION
**Pull request by pudelkoM**
Fixes #442

### CODE REVIEW

**Review by Yi-Tseng — APPROVED** [review]

File: stratum/hal/lib/common/openconfig_converter.cc [hunk 1]
```diff
@@ -410,10 +410,10 @@ ::util::StatusOr<Chassis> ComponentToChassis(
     case OPENCONFIGHERCULESPLATFORMPLATFORMTYPE_P4_SOFT_SWITCH:
       to.set_platform(PLT_P4_SOFT_SWITCH);
       break;
-    case OPENCONFIGHERCULESPLATFORMPLATFORMTYPE_BAREFOOT_TOFINO:
+    case OPENCONFIGHERCULESPLATFORMPLATFORMTYPE_GENERIC_BAREFOOT_TOFINO:
       to.set_platform(PLT_GENERIC_BAREFOOT_TOFINO);
       break;
-    case OPENCONFIGHERCULESPLATFORMPLATFORMTYPE_BAREFOOT_TOFINO2:
+    case OPENCONFIGHERCULESPLATFORMPLATFORMTYPE_GENERIC_BAREFOOT_TOFINO2:
```
Comment (bocon13) [inline stratum/hal/lib/common/openconfig_converter.cc:416]: What about the reverse direction?  There's a TODO from Yi, but I don't have the context for it.

File: stratum/hal/lib/common/openconfig_converter.cc [hunk 2]
```diff
@@ -396,24 +431,30 @@ ::util::StatusOr<Chassis> ComponentToChassis(
 
   switch (component.chassis().platform()) {
     case OPENCONFIGHERCULESPLATFORMPLATFORMTYPE_GENERIC_TRIDENT_PLUS:
-      to.set_platform(PLT_GENERIC_TRIDENT2);
```
Comment (pudelkoM) [inline stratum/hal/lib/common/openconfig_converter.cc:399]: lol

Comment (bocon13) [reply stratum/hal/lib/common/openconfig_converter.cc:399]: why are these all wrong?

**Review by bocon13 — APPROVED** [review]

**Review by bocon13 — COMMENTED** [review]
should we add a unit test?

**Comment by pudelkoM** [comment]
> should we add a unit test?

Since `ChassisToComponent` and `ComponentToChassis` are not publicly exposed, the only way the test them is through a full OC <-> ChassisConfig conversion. 

I created an issue for this: https://github.com/stratum/stratum/issues/451

**Pull request merged**